### PR TITLE
LaTeX: add latexlistlabels TeX boolean (for sphinxsetup)

### PR DESCRIPTION
**Pull request by jfbu**
Fixes: #6511

- Bugfix, no backwards incompatible change.

### CODE REVIEW

File: doc/latex.rst [hunk 1]
```diff
@@ -520,9 +520,19 @@ LaTeX boolean keys require *lowercase* ``true`` or ``false`` values.
...
+``latexlistlabels``
+    Defaults to ``false``.  It means that enumerated list items will follow
+    the format (alphabetical or numerical) from the source.  Setting to
+    ``true`` means to let LaTeX document class or dedicated package decide
+    of the style of enumeration according to nesting level.
+
+    .. versionadded:: 2.1.3
+       Use this via raw latex directive for local modification.
+       Use rather :confval:`latex_use_latexlistlabels` for a global setting.
+
```
Comment (jfbu) [inline doc/latex.rst:534]: I hesitated to document this (which is used anyhow by LaTeX writer for implementation of ``latex_use_latexlistlables`` confval), as it is of interest only for rare cases where one needs a local exception to global rule, and will be used only by LaTeX proficient people.

File: sphinx/texinputs/sphinx.sty [hunk 2]
```diff
@@ -413,6 +414,22 @@
 \newcommand\sphinxsetup[1]{\setkeys{sphinx}{#1}}
 
 
+%% ALPHANUMERIC LIST ITEMS
+\newcommand\sphinxsetlistlabels[5]
```
Comment (jfbu) [inline sphinx/texinputs/sphinx.sty:418]: I decided not to mention this macro in the documentation. But advanced LaTeX users can redefine it to their liking to customize labels of enumerated lists even without using any package.

File: sphinx/writers/latex.py [hunk 3]
```diff
@@ -629,6 +629,8 @@ def __init__(self, document, builder):
             self.elements['contentsname'] = self.babel_renewcommand('\\contentsname',
                                                                     contentsname)
 
+        if self.config.latex_use_latexlistlabels:
+            self.elements['sphinxpkgoptions'] += ',latexlistlabels'
```
Comment (jfbu) [inline sphinx/writers/latex.py]: ``latexlistlabels`` serves as synonym to ``latexlistlabels=true``. The default is ``false``.

File: sphinx/writers/latex.py [hunk 4]
```diff
@@ -1373,11 +1375,8 @@ def get_nested_level(node):
         suffix = node.get('suffix', '.')
 
         self.body.append('\\begin{enumerate}\n')
-        self.body.append('\\def\\the%s{%s{%s}}\n' % (enum, style, enum))
-        self.body.append('\\def\\label%s{%s\\the%s %s}\n' %
-                         (enum, prefix, enum, suffix))
-        self.body.append('\\makeatletter\\def\\p@%s{\\p@%s %s\\the%s %s}\\makeatother\n' %
-                         (enumnext, enum, prefix, enum, suffix))
+        self.body.append('\\sphinxsetlistlabels{%s}{%s}{%s}{%s}{%s}%%\n' %
+                         (style, enum, enumnext, prefix, suffix))
```
Comment (jfbu) [inline sphinx/writers/latex.py]: This is simply refactoring of original for shorter TeX file and potential customization hook.

File: doc/latex.rst (hunk 1 above)
Comment (jfbu) [reply doc/latex.rst:534]: Previous comment changes meaning with latest commit which dropped `latex_use_latexlistlabels`  config value in favour of keeping only the `sphinxsetup` TeX boolean interface.

**Review by jfbu — COMMENTED** [review]
Modified according to [this comment](https://github.com/sphinx-doc/sphinx/issues/6511#issuecomment-505327760) to drop the Python config value and keep only the underlying TeX boolean via `sphinxsetup` interface.

**Comment by jfbu** [comment]
squashed and rebased onto 2.0

**Comment by jfbu** [comment]
@tk0miya do you think I should withdraw this? original issue #6511 was closed via #6533

**Comment by tk0miya** [comment]
Yes, new option is no longer needed. #6533 resolves the problem well :-)

**Comment by tk0miya** [comment]
Thank you always!

**Pull request closed without merging**